### PR TITLE
new shell spec for unbricking our parachain

### LINKED ISSUE #65: collator authorities have been commented
https://github.com/integritee-network/parachain/blob/d0dac8b74201a50dfcc7caf798150d7cf7ff8908/polkadot-parachains/src/chain_spec.rs#L91-L103

not sure if this is what we registered on kusama.

Anyway: what we have registered on Kusama has been tested on westend. but it could be that only one of the collators was actually authoring all the blocks

### PR DESCRIPTION
**Pull request by brenzi**
closes #65
part of #77

* re-genesis all chain specs (Kusama will be reset, westend will be torn down in favor of a fresh rococo deployment)
* de-clutter spec generation. added a few improvements from encointer too
* bump version to 1.5.13
* 
this PR shall build the basis of the to-be-reset genesis config for Kusama

tested
* [x] rococo-local-with-shell and upgrade to full runtime
* [x] kusama-local-with-shell (need to wait 60min before producing blocks!!) runtime upgrade only takes seconds after that
* [x] launch without relay `--chain integritee-kusama` 
   * paraid correct for parachainInfo.parachainId = 2015
   * sudo key correct
   * sudo balance 10 TEER

### CODE REVIEW

File: polkadot-parachains/src/chain_spec.rs [hunk 1]
```diff
@@ -92,13 +92,14 @@ impl IntegriteeKeys {
...
 			public_from_ss58::<sr25519::Public>("5GZJjbPPD9u6NDgK1ApYmbyGs7EBX4HeEz2y2CD38YJxjvQH")
 				.into(),
-			/*
-			public_from_ss58::<sr25519::Public>("5CcSd1GZus6Jw7rP47LLqMMmtr2KeXCH6W11ZKk1LbCQ9dPY").into(),
-			public_from_ss58::<sr25519::Public>("5FsECrDjBXrh5hXmN4PhQfNPbjYYwwW7edu2UQ8G5LR1JFuH").into(),
-			public_from_ss58::<sr25519::Public>("5HBdSEnswkqm6eoHzzX5PCeKoC15CCy88vARrT8XMaRRuyaE").into(),
-			public_from_ss58::<sr25519::Public>("5GGxVLYTXS7JZAwVzisdXbsugHSD6gtDb3AT3MVzih9jTLQT").into(),
-
-			 */
+			public_from_ss58::<sr25519::Public>("5CcSd1GZus6Jw7rP47LLqMMmtr2KeXCH6W11ZKk1LbCQ9dPY")
```
Comment (brenzi) [inline polkadot-parachains/src/chain_spec.rs:95]: this fixes #65

File: polkadot-parachains/src/command.rs [hunk 2]
```diff
@@ -42,7 +41,11 @@ use sp_core::hexdisplay::HexDisplay;
 use sp_runtime::traits::Block as BlockT;
 use std::{io::Write, net::SocketAddr};
 
-const DEFAULT_PARA_ID: u32 = 2015;
+const LOCAL_PARA_ID: u32 = 2000;
```
Comment (brenzi) [inline polkadot-parachains/src/command.rs:44]: this was overdue. as we can't select our paraid ourselves, here are all the reserved id's

File: scripts/update_hardcoded_specs.py [hunk 3]
```diff
@@ -21,26 +21,20 @@
...
 SPECS = [
-    {
-        "chain_id": "shell-kusama",
-        "para_id": 2015,
-    },
-    {
-        "chain_id": "integritee-kusama",
-        "para_id": 2015,
-    }
+    "integritee-rococo",
```
Comment (brenzi) [inline scripts/update_hardcoded_specs.py:24]: paraid not needed anymore

File: scripts/update_hardcoded_specs.py [hunk 4]
```diff
@@ -70,15 +64,16 @@ def main(migrate_genesis: bool):
...
         os.remove(f'{new_file_base}-raw.json')
+        os.remove(f'{new_file_base}-raw-unsorted.json')
         os.remove(f'{new_file_base}.state')
         os.remove(f'{new_file_base}.wasm')
 
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument('--migrate-genesis', help='Create entirely new chain spec, not preserving previous genesis state', action='store_true')
+    parser.add_argument('--regenesis', help='Overwrite genesis state in chain spec. Use this for resetting chains entirely', action='store_true')
```
Comment (brenzi) [inline scripts/update_hardcoded_specs.py:74]: this I just copied over from encointer which fixed the misunderstanding of the flag

File: polkadot-parachains/src/command.rs [hunk 5]
```diff
@@ -65,40 +68,43 @@ impl<T: sc_service::ChainSpec + 'static> IdentifyChain for T {
...
 ) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-	// FIXME distinguish paraid by relaychain
-	let para_id: ParaId = DEFAULT_PARA_ID.into();
 	Ok(match id {
-		"integritee-rococo-local" => Box::new(integritee_chain_spec(para_id, GenesisKeys::Integritee, RelayChain::RococoLocal)),
-		"integritee-rococo-local-dev" => Box::new(integritee_chain_spec(para_id, GenesisKeys::WellKnown, RelayChain::RococoLocal)),
-		"integritee-rococo-fresh" => Box::new(integritee_chain_spec(para_id, GenesisKeys::Integritee, RelayChain::Rococo)),
-		"integritee-rococo" => Box::new(integritee_rococo_config()?),
+		// live configs (hard coded genesis state. genesis will always be shell for a live system)
+		"integritee-rococo" => Box::new(shell_rococo_config()?),
```
[the remark this answers is not in the document]
Comment (brenzi) [reply polkadot-parachains/src/command.rs:73]: @mosonyi this also means our live deployment will always use `--chain integritee-kusama` after our upgrade

[the remark this answers is not in the document]
Comment (mosonyi) [reply polkadot-parachains/src/command.rs:73]: OK, noted.

File: polkadot-parachains/src/chain_spec.rs [hunk 6]
```diff
@@ -92,13 +92,14 @@ impl IntegriteeKeys {
...
+				.into(),
+			public_from_ss58::<sr25519::Public>("5FsECrDjBXrh5hXmN4PhQfNPbjYYwwW7edu2UQ8G5LR1JFuH")
+				.into(),
+			public_from_ss58::<sr25519::Public>("5HBdSEnswkqm6eoHzzX5PCeKoC15CCy88vARrT8XMaRRuyaE")
+				.into(),
+			public_from_ss58::<sr25519::Public>("5GGxVLYTXS7JZAwVzisdXbsugHSD6gtDb3AT3MVzih9jTLQT")
+				.into(),
```
Comment (brenzi) [inline polkadot-parachains/src/chain_spec.rs:102]: this fixes #65

File: polkadot-parachains/src/command.rs [hunk 7]
```diff
@@ -42,7 +41,11 @@ use sp_core::hexdisplay::HexDisplay;
 use sp_runtime::traits::Block as BlockT;
 use std::{io::Write, net::SocketAddr};
 
-const DEFAULT_PARA_ID: u32 = 2015;
+const LOCAL_PARA_ID: u32 = 2000;
+const ROCOCO_PARA_ID: u32 = 2073;
+const WESTEND_PARA_ID: u32 = 2081;
+const KUSAMA_PARA_ID: u32 = 2015;
+const POLKADOT_PARA_ID: u32 = 2039;
```
Comment (brenzi) [inline polkadot-parachains/src/command.rs:48]: this was overdue. as we can't select our paraid ourselves, here are all the reserved id's

File: polkadot-parachains/src/command.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (clangenb) [reply polkadot-parachains/src/command.rs:73]: Yeah, I agree. The integritee-hardcoded spec was essentially useless...

**Review by clangenb — COMMENTED** [review]
Did not test, but looks good to me.

**Pull request merged**